pkg/util/taskset/task_set_test.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package taskset
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTaskSetSingleWorker(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// When a single worker claims tasks from a taskSet, it should claim all
+	// tasks in sequential order now that we use FIFO claiming.
+	tasks := MakeTaskSet(10, 1)
+	var found []TaskID
+
+	for next := tasks.ClaimFirst(); !next.IsDone(); next = tasks.ClaimNext(next) {
+		found = append(found, next)
+	}
+
+	// Verify that tasks are claimed sequentially.
+	require.Equal(t, []TaskID{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, found)
```

Comment:
Neat! It's cool to see this case got improved while keeping the same overall API.